app.js

```diff
@@ -35,41 +37,112 @@ addMushroomButton.addEventListener('click', () => {
 
         mushroomCount++;
         displayMushrooms();
+    } else if (Math.random() > .5) {
+        alert('found a strawberry!');
+        berryCount++;
+        displayMushrooms();
     } else {
         alert('no luck!');
     }
+    console.log(berryCount);
 });
 
 addFriendButton.addEventListener('click', () => {
+
     // get the name from the input
+    const data = friendInputEl.value;
+    // create a new friend object   
+    const newFriend = {
+        name: data || `Friend# ${Math.floor(Math.random() * 1000)}`,
+        satisfaction: Math.floor(Math.random() * 2 + 1),
+    };
+    // push it into the friends state array, passed in as an argument
+    friendData.push(newFriend);
+    // reset the input
+    // display all the friends (use a function here)
+    displayFriends();
+});
+
+friendForm.addEventListener('submit', (event) => {
```

Comment:
This works when I hit enter in the input but not when I click the Invite a new friend button. Probably best to just stick to one or the other. 

This one does clear out the input field on submit which is awesome. The above event listener for the button click doesn't clear out the input field so if you keep clicking it just submits the same name over and over.